web/src/containers/PatientDetails/PatientDocument/index.tsx

```diff
@@ -56,18 +64,25 @@ export function PatientDocument(props: Props) {
                 <RenderRemoteData remoteData={response} renderLoading={Spinner}>
                     {({ formData, onSubmit, provenance }) => (
                         <>
-                            <div className={s.header}>
-                                <Title level={3}>{formData.context.questionnaire.name}</Title>
-                            </div>
+                            <PatientDocumentHeader
+                                formData={formData as QuestionnaireResponseFormData}
+                                questionnaireId={questionnaireId}
+                                draftSaveState={draftSaveState}
+                                savedMessage={savedMessage}
+                            />
+
                             <BaseQuestionnaireResponseForm
-                                formData={formData}
+                                formData={formData as QuestionnaireResponseFormData}
```

Comment:
I don't see any need of use `as QuestionnaireResponseFormData`. I think it can be removed from all the places in this file